### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by gutaussehend-Harshal**

### CODE REVIEW

File: Scripts/ButtonController.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
```
Comment (ahujarahul99) [inline Scripts/ButtonController.cs:3]: NameSpaces in the class.

File: Scripts/ButtonController.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+
+public class ButtonController : MonoBehaviour
+{
+    public Button Backbutton;
```
Comment (ahujarahul99) [inline Scripts/ButtonController.cs:6]: Camel Naming convention should be follow here.

File: Scripts/ButtonController.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+public class ButtonController : MonoBehaviour
+{
+    public Button Backbutton;
+    public GameObject LevelSeletion;
+    void Start()
+    {
+        Backbutton.onClick.AddListener(ButtonClick);
+    }
+    void ButtonClick()
```
Comment (ahujarahul99) [inline Scripts/ButtonController.cs:12]: Summarise the function give the comment for the same .

File: Scripts/ButtonController.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    {
+        // SceneManager.LoadScene("Level1");
+        LevelSeletion.SetActive(true);
+        Backbutton.gameObject.SetActive(false);
```
Comment (ahujarahul99) [inline Scripts/ButtonController.cs:16]: Follow camel Naming conventions .

File: Scripts/CameraController.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
```
Comment (ahujarahul99) [inline Scripts/CameraController.cs:4]: Add NameSpaces

File: Scripts/CameraController.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour
+{
+    public Transform target;
```
Comment (ahujarahul99) [inline Scripts/CameraController.cs:7]: Use Serialized field when we need something in the inspector that is class specific.

File: Scripts/CameraController.cs [hunk 7]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour
+{
+    public Transform target;
+    public float smoothSpeed = 0.125f;
```
Comment (ahujarahul99) [inline Scripts/CameraController.cs:8]: Serialize field

File: Scripts/CameraController.cs [hunk 8]
```diff
@@ -0,0 +1,28 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CameraController : MonoBehaviour
+{
+    public Transform target;
+    public float smoothSpeed = 0.125f;
+
+    public Vector3 offset;
```
Comment (ahujarahul99) [inline Scripts/CameraController.cs:10]: Serialized field  and make it private when its class specific.

File: Scripts/CameraController.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    public Transform target;
+    public float smoothSpeed = 0.125f;
+
+    public Vector3 offset;
+
+    // [SerializeField]
+    // private float offSetX;
+    // [SerializeField]
+    // private float offSetY;
+    void FixedUpdate()
```
Comment (ahujarahul99) [inline Scripts/CameraController.cs:16]: Just try once this with the late update.

File: Scripts/EndGame.cs [hunk 10]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
```
Comment (ahujarahul99) [inline Scripts/EndGame.cs:6]: Give namespace and give a meaningful name to the class.

File: Scripts/EndGame.cs [hunk 11]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+public class EndGame : MonoBehaviour
+{
```
Comment (ahujarahul99) [inline Scripts/EndGame.cs:8]: Give a specification what exactly you are doing in this function.

File: Scripts/EnemyController.cs [hunk 12]
```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyController : MonoBehaviour
+{
+    public float speed;
```
Comment (ahujarahul99) [inline Scripts/EnemyController.cs:7]: Add serialize field  here with private variable.

File: Scripts/EnemyController.cs [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+
+public class EnemyController : MonoBehaviour
+{
+    public float speed;
+    private bool moveRight;
+    void Update()
+    {
+        if (moveRight)
+        {
+            transform.Translate(2 * Time.deltaTime * speed, 0, 0);
```
Comment (ahujarahul99) [inline Scripts/EnemyController.cs:13]: Do not use 2or 4 multiplication do that same in speed .

File: Scripts/EnemyController.cs [hunk 14]
```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EnemyController : MonoBehaviour
+{
```
Comment (ahujarahul99) [inline Scripts/EnemyController.cs:6]: Try with [SerializeField] private Vector2 enemyScale.

File: Scripts/EnemyController.cs [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+    void Update()
+    {
+        if (moveRight)
+        {
+            transform.Translate(2 * Time.deltaTime * speed, 0, 0);
+            transform.localScale = new Vector2(2, 2);
+        }
+        else
+        {
+            transform.Translate(-2 * Time.deltaTime * speed, 0, 0);
```
Comment (ahujarahul99) [inline Scripts/EnemyController.cs:18]: Here also use variables not -2 .

File: Scripts/EnemyController.cs [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    private void OnTriggerEnter2D(Collider2D trig)
+    {
+        if (trig.gameObject.CompareTag("turn"))
+        {
+            if (moveRight)
+            {
+                moveRight = false;
+
```
Comment (ahujarahul99) [inline Scripts/EnemyController.cs:30]: Dont give free spaces it will impact readability of the code.

File: Scripts/GameOverController.cs [hunk 17]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
```
Comment (ahujarahul99) [inline Scripts/GameOverController.cs:6]: Namespaces and summarise the class .

File: Scripts/GameOverController.cs [hunk 18]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+public class GameOverController : MonoBehaviour
+{
+    public Button buttonRestart;
```
Comment (ahujarahul99) [inline Scripts/GameOverController.cs:9]: Use serialize field and make it private and use it via inspector.

File: Scripts/GameOverController.cs [hunk 19]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        buttonRestart.onClick.AddListener(ReloadLevel);
+    }
+    public void PlayerDied()
+    {
+        gameObject.SetActive(true);
+    }
+    public void ReloadLevel()
+    {
+        SceneManager.LoadScene("Level1");
```
Comment (ahujarahul99) [inline Scripts/GameOverController.cs:21]: Give the name of the scene in a string variable.

File: Scripts/KeyController.cs [hunk 20]
```diff
@@ -0,0 +1,16 @@
...
+
+public class KeyController : MonoBehaviour
+{
+    private void OnCollisionEnter2D(Collision2D collision)
+    {
+        if(collision.gameObject.GetComponent<PlayerController>() != null) 
+        {
+            PlayerController playerController = collision.gameObject.GetComponent<PlayerController>();
+            playerController.PickUpKey();
+            Destroy(gameObject);
```
Comment (ahujarahul99) [inline Scripts/KeyController.cs:13]: Instead of Destroying use object pooling

File: Scripts/Level/LevelLoader.cs [hunk 21]
```diff
@@ -0,0 +1,33 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+[RequireComponent(typeof(Button))]
+public class LevelLoader : MonoBehaviour
+{
+    private Button button;
+    public string LevelName;
```
Comment (ahujarahul99) [inline Scripts/Level/LevelLoader.cs:9]: Use serialize field with private variable  here and use camel naming conventions.

File: Scripts/Level/LevelManager.cs [hunk 22]
```diff
@@ -0,0 +1,63 @@
+﻿using UnityEngine;
+using System;
+using UnityEngine.SceneManagement;
+public class LevelManager : MonoBehaviour
+{
+    private static LevelManager instance;
+    // public string Level1;
+    public string[] Levels;
```
Comment (ahujarahul99) [inline Scripts/Level/LevelManager.cs:8]: Here you can also use serialize fields.

File: Scripts/Level/LevelManager.cs [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public void MarkCurrentLevelComplete()
+    {
+        Scene currentScene = SceneManager.GetActiveScene();
+
+        // Set level status to complete
+        SetLevelStatus(currentScene.name, LevelStatus.Completed);
+
+        // Unlock the next level
```
Comment (ahujarahul99) [inline Scripts/Level/LevelManager.cs:38]: Dead code should not be there .

File: Scripts/Level/LevelStatus.cs [hunk 24]
```diff
@@ -0,0 +1,16 @@
+// using System;
+// public class Levels
+// {
```
Comment (ahujarahul99) [inline Scripts/Level/LevelStatus.cs:3]: Remove this code.

File: Scripts/LobbyButtonController.cs [hunk 25]
```diff
@@ -0,0 +1,18 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+public class LobbyButtonController : MonoBehaviour
+{
+    public Button Quitbutton;
```
Comment (ahujarahul99) [inline Scripts/LobbyButtonController.cs:9]: Serialized the button

File: Scripts/PlayerController.cs [hunk 26]
```diff
@@ -0,0 +1,173 @@
+﻿using System.Collections;
```
Comment (ahujarahul99) [inline Scripts/PlayerController.cs:1]: Change this script as i have taught.

File: Scripts/PlayerController.cs [hunk 27]
```diff
@@ -0,0 +1,173 @@
...
+        // Destroy(gameObject);
+        // Play the death animation
+        // Reset Entire level
+
+        // Debug.Log("before" + livesRemain.ToString());
+        livesRemain--;
+        // Debug.Log("After" + livesRemain.ToString());
+        // animator.SetBool("Hurt", true);
+
+        transform.position = new Vector2(0, 0);
```
Comment (ahujarahul99) [inline Scripts/PlayerController.cs:56]: Vector2.zero.

File: Scripts/PlayerController.cs [hunk 28]
```diff
@@ -0,0 +1,173 @@
...
+    public float crouchoffsetY;
+    public float crouchsizeX;
+    public float crouchsizeY;
+    public float offsetX;
+    public float offsetY;
+    public float sizeX;
+    public float sizeY;
+
+
+    private int livesRemain = 3;
```
Comment (ahujarahul99) [inline Scripts/PlayerController.cs:26]: Update this code .

File: Scripts/PlayerController.cs [hunk 29]
```diff
@@ -0,0 +1,173 @@
...
+            gameOverController.PlayerDied();    
+        }
+        // ReloadLevel();
+        // this.enabled = false;
+    }
+    private void updateLifeUI()
+    {
+        if (livesRemain == 3)
+        {
+            life01.gameObject.SetActive(true);
```
Comment (ahujarahul99) [inline Scripts/PlayerController.cs:72]: Do not use life01 variables use specific names

**Comment by gutaussehend-Harshal** [comment]
Done

**Pull request closed without merging**